pytorch_blade/torch_blade/quantization/prepare_data.py

```diff
@@ -0,0 +1,247 @@
+            list_unpack.addInput(call_method.output())
+            list_unpack.eraseOutput(0)
+            for out in g_node.output_list():
+                lu_out = list_unpack.addOutput()
+                lu_out.setType(out.type())
+                out.replaceAllUsesWith(lu_out)
+
+            g_node.destroy()
+
+        self.get_attr_nodes_for_fusion_group = get_attr_nodes_for_fusion_group
+
+    def insert_observer(self):
+        """
+           Insert the DataCollectObserver to the graph. For convenience, we
+           group all inputs of each fusion group into a tensor list. After this function,
+           the graph is like:
+                graph(%self.1 : __torch__.___torch_mangle_2.MyModel,
+                      %x : Tensor,
+                      %y : Tensor,
+                      %5 : __torch__.torch.jit._script.ScriptModule):
+                    %6 : __torch__.libtorch_executable_fusion_group_0 = prim::GetAttr[name="libtorch_executable_fusion_group_0"](%5) # noqa
+
+                    %11 : Tensor[] = prim::ListConstruct(%x, %y)  <-- group all inputs
+                    %12 : DataCollectObserver = prim::GetAttr[name="collect_observer_0"](%5)  <-- get the observer
+                    %13 : Tensor[] = prim::CallMethod[name="forward"](%12, %11)  <-- call the observer
+                    %14 : Tensor, %15 : Tensor = prim::ListUnpack(%13)  <-- unpack the output of the observer
+
+                    %7 : Tensor[] = prim::CallMethod[name="forward"](%6, %14, %15)
+                    %9 : Tensor, %10 : Tensor = prim::ListUnpack(%7)
+                    return (%9, %10)
+        """
+        all_collect_observer_name = []
+        for idx, get_attr in enumerate(self.get_attr_nodes_for_fusion_group):
+            users = [u.user for u in get_attr.output().uses()]
+            if len(users) != 1:
+                raise RuntimeError("The output of prim::GetAttr of each executable should "
+                                   "be used by only one .")
+
+            # Combine the inputs of the Fusion Group into a list
+            # and make the observer record the list
+            forward_node = users[0]
+            list_pack = self.graph.create('prim::ListConstruct')
+            self.graph.appendNode(list_pack)
+            list_pack.moveBefore(forward_node)
+            for inp in forward_node.input_list()[1:]:
+                list_pack.addInput(inp)
+            list_pack.output().setType(torch_blade.tools.get_list_tensor_type())
+
+            # insert the observer, three steps:
+            #   1. register the observer
+            #   2. create getattr node for the observer
+            #   3. create call method node for the observer
+            collect_observer = DataCollectObserver()
+            collect_observer = torch.jit.script(collect_observer)
+            observer_name = f"{self.collect_observer_prefix}_{idx}"
+            all_collect_observer_name.append(observer_name)
```

Comment:
So did here, we can reuse the similar codes written previously. Please refactor the code and move it to the right commonplace if it's needed.

Reply:
done, I reuse the codes about inserting the observer